shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/PlantsRepository.kt

```diff
@@ -1,43 +1,38 @@
 package com.vml.tutorial.plantshop.plants.data
 
-import com.vml.tutorial.plantshop.plants.domain.DbDataSource
+import com.vml.tutorial.plantshop.plants.domain.FavoritesDataSource
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
-import com.vml.tutorial.plantshop.plants.presentation.PlantCategory
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 interface PlantsRepository {
-    fun getPlants(category: PlantCategory): List<Plant>
+    suspend fun getPlants(): List<Plant>?
     fun getFavorites(): Flow<List<Plant>>
     suspend fun toggleFavoriteStatus(plantId: Int)
 }
 
 class PlantsRepositoryImpl(
-    private val localDataSource: PlantsDataSource,
-    private val dbDataSource: DbDataSource
+    private val dbPlantsDataSource: PlantsDataSource,
+    private val favoritesDataSource: FavoritesDataSource
 ) : PlantsRepository {
-    override fun getPlants(category: PlantCategory): List<Plant> {
-        return localDataSource.getPlants().filter { plant ->
-            plant.types.contains(category.type)
-        }
+    override suspend fun getPlants(): List<Plant>? {
+        return dbPlantsDataSource.getPlants()
```

Comment:
Check here our local db: if it contains plants than return with those, if not than fetch it from remote data source (firebase), store it into local db and return the results :)

Reply:
Done